core_types/src/dna/bridges.rs

```diff
@@ -0,0 +1,62 @@
+use crate::cas::content::Address;
+
+/// A bridge is the definition of a connection to another DNA that runs under the same agency,
+/// i.e. in the same container.
+///
+/// Defining a bridge means that the code in this DNA can call zome functions of that other
+/// DNA.
+///
+/// The other DNA can either be referenced statically by exact DNA address/hash or dynamically
+/// by defining the trait that other DNA has to implement in order to be used as bridge.
+///
+/// Bridges can be required or optional. If a required bridge DNA is not installed this DNA
+/// can't run, so required bridges are hard dependencies that have to be enforced by the container.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Hash)]
+#[serde(untagged)]
+pub enum Bridge {
+    Address(AddressBridge),
```

Comment:
I as I said on the heartbeat today I do think it's better to have the `dna_address` be the enum (e.g. `DnaReference`), not the `Bridge` struct itself, but I'm ok for that to be a follow-up

Reply:
Yeah, I almost did this change just now but then I thought I would love to get the details straight on the function groups (=capabilities) which are changes to the DNA as well.. can we talk about that before finalizing this one?

Reply:
done